includes/filename.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+// Exit if accessed directly
+if(!defined('ABSPATH')){
+	exit;
+}
+
+/**
+ * Cleanup file name
+ */
+class Fiber_Admin_Filename{
+	public function __construct(){
+		// Cleanup file name
+		add_filter('sanitize_file_name_chars', [$this, 'fiad_special_chars']);
+		add_filter('sanitize_file_name', [$this, 'fiad_cleanup_file_name'], 10, 2);
+	}
+	
+	/*
+	 * Return fiad special chars
+	 */
+	public function fiad_special_chars($special_chars){
+		if(($key = array_search('%', $special_chars)) !== false){
+			unset($special_chars[$key]);
+		}
+		
+		return $special_chars;
+	}
+	
+	public function fiad_handle_special_chars($sanitized_filename){
+		$sanitized_filename = preg_replace('/[^A-Za-z0-9- ]/', '-', $sanitized_filename); // Remove special char not specified default by WordPress
+		$sanitized_filename = preg_replace('/-{2,}/', '-', $sanitized_filename); // Replace a row of - with only 1 -
+		$sanitized_filename = trim($sanitized_filename, '-'); // Remove - at the start
+		
+		// Remove - at the end
+		return rtrim($sanitized_filename, '-');
```

Comment:
I don't think we need `rtrim` here since `trim` function already does the job, read the manual of `trim` function here: https://www.php.net/manual/en/function.trim.php.

Reply:
I didn't read the docs carefully, I thought it only trims from the beginning, thank you so much for pointing out my mistake!